### PR TITLE
Added shared/oval/is_a_container.xml to further enable SSG

### PR DESCRIPTION
**Pull request by rsprudencio**
to tell if scan target is a container or not

### CODE REVIEW

**Comment by rsprudencio** [comment]
@redhatrises sure!

**Comment by redhatrises** [comment]
Thanks! Ack.

**Comment by redhatrises** [comment]
@rsprudencio The el7 Jenkins node is offline. Not sure if you can take a look and fix it.

File: shared/oval/installed_env_is_a_container.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<def-group>
+  <definition class="compliance" id="is.container" version="2">
+    <metadata>
+      <title>Check if the scan target is a container</title>
```
Comment (yuumasato) [inline shared/oval/installed_env_is_a_container.xml]: `Check if the scan target is a Docker container`?

File: shared/oval/installed_env_is_a_container.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      <description>Check if file /.dockerenv exists, if it does then we consider to be a docker filesystem.</description>
+    </metadata>
+    <criteria>
+      <criterion comment="Check if /.dockerenv exists" test_ref="is.container" />
+    </criteria>
+  </definition>
+
+  <unix:file_test check="all" check_existence="all_exist" comment="Check if /.dockerenv exists" id="is.container" version="1">
```
Comment (yuumasato) [inline shared/oval/installed_env_is_a_container.xml]: `is.docker_container`?

File: shared/oval/installed_env_is_a_container.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    <unix:object object_ref="is.container" />
+  </unix:file_test>
+
+  <unix:file_object comment="Check file /.dockerenv" id="is.container" version="1">
```
Comment (yuumasato) [inline shared/oval/installed_env_is_a_container.xml]: `is.docker_container`?

**Comment by redhatrises** [comment]
The other thing is that the definition class id should match the name of the file.
`<definition class="compliance" id="is.container" version="2">` should be <definition class="compliance" id="installed_env_is_a_container" version="2">

File: shared/oval/installed_env_is_a_container.xml (hunk 3 above)
Comment (mpreisler) [inline shared/oval/installed_env_is_a_container.xml]: I would keep is.container and add more definitions when we need to add support for more tech.

**Comment by mpreisler** [comment]
`class="compliance"` is the wrong class.

**Comment by mpreisler** [comment]
Before merging this I think we should have the CPE names in place as well. And the CPE names should be using the new OVAL definitions.

**Comment by rsprudencio** [comment]
@mpreisler should I use "inventory" class?

**Comment by rsprudencio** [comment]
@yuumasato it is "docker specific" for now but not because it should be, only because it's the only way we can tell the filesystem is a container for now, we should support most container technologies but Docker is a start So as @mpreisler stated, I also think this shouldn't be named after Docker as it is supposed to be tech-agnostic.

**Comment by rsprudencio** [comment]
Linked to #1645. Also proposed another CPE cpe:/a:machine so we can also flag Bare-metal/VM-only rules.

**Comment by mpreisler** [comment]
ACK

**Pull request merged**